search/includes/classes/class-search.php

```diff
@@ -948,16 +957,26 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 
 		$timeout = $this->get_http_timeout_for_query( $query, $args );
 
+		if ( $is_testing_next_version ) {
+			$url = $this->build_elasticsearch_url( $query['url'], self::ELASTICSEARCH_MIGRATION_NEXT );
+
+			if ( ! headers_sent() ) {
+				header( sprintf( 'X-Elasticsearch-Version: %s', self::ELASTICSEARCH_MIGRATION_NEXT ) );
+			}
```

Comment:
```suggestion
			add_action( 'send_headers', fn() => header( sprintf( 'X-Elasticsearch-Version: %s', self::ELASTICSEARCH_MIGRATION_NEXT ) ) );
```
WP-idiomatic way would be to use `send_headers` hook. NBD but something to be aware of.